packages/runtime/src/ctx.ts

```diff
@@ -150,8 +150,14 @@ export class Ctx implements Context {
     const toAssetHolding = this.getAssetHolding(txnParam.assetID, txnParam.toAccountAddr);
     txnParam.amount = BigInt(txnParam.amount);
 
-    this.assertAssetNotFrozen(txnParam.assetID, fromAccountAddr);
-    this.assertAssetNotFrozen(txnParam.assetID, txnParam.toAccountAddr);
+    if (txnParam.payFlags.closeRemainderTo !== undefined) {
+      this.assertAssetNotFrozen(txnParam.assetID, txnParam.payFlags.closeRemainderTo);
+    }
+    if (txnParam.amount > 0) {
+      this.assertAssetNotFrozen(txnParam.assetID, fromAccountAddr);
+      this.assertAssetNotFrozen(txnParam.assetID, txnParam.toAccountAddr);
+    }
+
```

Comment:
is this one also verified from algorand docs ? Please add a link above.

Reply:
@amityadav0 Couldn't find any proper documentation on this. I think the checks I have made make pretty good sense.

Reply:
okay, please add comments describing the checks.